src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -72,6 +72,10 @@ public sealed class ConnectionString
         private TimeSpan? _connectTimeout;
         private string _databaseName;
         private bool? _fsync;
+        private bool _hasSslBeenSet;
+        private bool _hasTlsAllowInvalidCertificatesBeenSet;
+        private bool _hasTlsAllowInvalidHostnamesBeenSet;
+        private bool _hasTlsInsecureBeenSet;
```

Comment:
These fields don't seem right. Seems like it should be:

    private bool? _tls;
    private bool? _tlsAllowInvalidCertificates;
    private bool? _tlsAllowInvalidHostnames;
    private bool? _tlsInsecure;

Add new public properties for these fields also.